### PR TITLE
Yq add optional image param

### PR DESCRIPTION
**Pull request by frerikandriessen**
# Changes

* Added an `image` param so that you have more control over which version of yq you want to use
* Updated the default value to the newest image
* Added the `set -e` flag so that the script does not report success when one of the commands failed
# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [X] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [X] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [X] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [X] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [X] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [X] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [X] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [X] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [X] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

**Comment by GijsvanDulmen** [comment]
Nice contribution! :-)

File: task/yq/0.3/yq.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      type: string
+      description: The yq image to use.
+      default: docker.io/mikefarah/yq:4.16.2@sha256:0d4f6e27bdcac7316f635acd524ab0eecc4ad50834b54d10322268650c7712cb
+  steps:
+    - name: replace-in-yaml
+      image: $(params.image)
+      workingDir: $(workspaces.source.path)
+      args: ["$(params.files[*])"]
+      script: |
+        /usr/bin/env sh
```
Comment (06kellyjac) [inline task/yq/0.3/yq.yaml:36]: is this meant to be a shebang?

`#!/usr/bin/env sh`

Comment (frerikandriessen) [reply task/yq/0.3/yq.yaml:36]: Yes.. good catch!

**Pull request merged**